server/src/main/java/org/elasticsearch/cluster/metadata/DataLifecycle.java

```diff
@@ -148,16 +127,16 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(dataRetention);
+        return Objects.hash(dataRetention);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeOptionalTimeValue(dataRetention);
+        out.writeOptionalWriteable(dataRetention);
```

Comment:
I think we need to make these changes in a BWC way and add version checks here

Reply:
You are right, I did that and reverted the version bump in the tests.